backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/serialization/KonanIr.proto

```diff
@@ -159,30 +236,32 @@ message IrGetClass {
 }
 
 message IrGetEnumValue {
-    required KotlinType type = 1;
-    required KotlinDescriptor descriptor = 2;
+    required IrType type = 1;
+    required IrSymbol symbol = 2;
 }
 
 message FieldAccessCommon {
-    required KotlinDescriptor descriptor = 1;
-    optional KotlinDescriptor super = 2;
-    required IrExpression receiver = 3;
+    required IrSymbol symbol = 1;
+    optional IrSymbol super = 2;
+    optional IrExpression receiver = 3;
 }
 
 message IrGetField {
     required FieldAccessCommon field_access = 1;
+    required IrType type = 2;
 }
 
 message IrGetValue {
-    required KotlinDescriptor descriptor = 1;
+    required IrSymbol symbol = 1;
+    required IrType type = 2;
```

Comment:
Does it duplicate `IrExpression.type`?

Reply:
indeed. Here and in several other places.